thrift/shared.thrift

```diff
@@ -1713,4 +1713,100 @@ struct RefreshWorkflowTasksRequest {
 
 struct FeatureFlags {
 	10: optional bool WorkflowExecutionAlreadyCompletedErrorEnabled
-}
\ No newline at end of file
+}
+
+enum CrossClusterTaskType {
+  StartChildExecution
+  CancelExecution
+  SignalExecution
+}
+
+enum CrossClusterTaskFailedCause {
+  DOMAIN_NOT_ACTIVE
+  DOMAIN_NOT_EXISTS
+  WORKFLOW_ALREADY_RUNNING
+  WORKFLOW_NOT_EXISTS
+}
+
+struct CrossClusterTaskInfo {
+  10: optional string domainID
+  20: optional string workflowID
+  30: optional string runID
+  40: optional CrossClusterTaskType taskType
+  50: optional i16 taskState
+  60: optional i64 (js.type = "Long") taskID
+  70: optional i64 (js.type = "Long") visibilityTimestamp
+}
+
+struct CrossClusterStartChildExecutionRequestAttributes {
+  10: optional string targetDomainID
+  20: optional string requestID
+  30: optional i64 (js.type = "Long") initiatedEventID
+  40: optional StartChildWorkflowExecutionInitiatedEventAttributes initiatedEventAttributes
```

Comment:
Do you need the run ID when the task is processed as create first decision in child workflow?

Reply:
Thanks for catching. I've added `targetRunID` to this struct and a comment saying it for scheduling the first decision.